### PR TITLE
[BUGFIX beta] Removes `get record` from InternalModel.

### LINKED ISSUE #4756: Check use of `internalModel.record`
`IntenalModel` has a `get record` as well as a `getRecord` which is a bit confusing.

Likely we can remove the getter and replace it with `InternalModel.hasRecord` but this remains to be verified.

### PR DESCRIPTION
**Pull request by esbanarango**
This PR removes `get record` from InternalModel 'cause there is already a `getRecord` method.

Fixes #4756

### CODE REVIEW

File: addon/-private/system/references/record.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ RecordReference.prototype.push = function(objectOrPromise) {
    @return {DS.Model} the record for this RecordReference
 */
 RecordReference.prototype.value = function() {
-  return this.internalModel.record;
+  return this.internalModel._record;
```
Comment (sduquej) [inline addon/-private/system/references/record.js]: `getRecord()`?

Comment (bmac) [inline addon/-private/system/references/record.js]: @esbanarango do you mind changing this line to use `getRecord()`?

Comment (esbanarango) [inline addon/-private/system/references/record.js]: @bmac changed. 👍

**Comment by bmac** [comment]
Thanks @esbanarango.

**Pull request merged**